### PR TITLE
bzip3: add recipe

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **bzip3/1.1.4**

Bzip3 is a compression library.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/bzip3/all/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,59 @@
+cmake_minimum_required(VERSION 3.1)
```
Comment (SSE4) [inline recipes/bzip3/all/CMakeLists.txt:1]:
```suggestion
cmake_minimum_required(VERSION 3.4)
```

Comment (toge) [reply recipes/bzip3/all/CMakeLists.txt]: @SSE4 
Thank you!

**Review by danimtb — DISMISSED** [review]

File: recipes/bzip3/all/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    source_subfolder/src/libsais.c
+    source_subfolder/src/lzp.c
+    source_subfolder/src/rle.c
+)
+
+set(BZIP3_INC
+    source_subfolder/include/libbz3.h
+)
+
+include_directories(source_subfolder/include)
```
Comment (jwillikers) [inline recipes/bzip3/all/CMakeLists.txt:23]: Should the `target`-based from of `include_directories(source_subfolder/include)` be preferred here, i.e.  `target_include_directories(bzip3 PUBLIC source_subfolder/include)`?

File: recipes/bzip3/all/test_package/test_package.c [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#include <stdio.h>
+#include <string.h>
+
+#include "libbz3.h"
+
+int main() {
+  unsigned int block_size = 8 * 1024 * 1024;
```
Comment (jwillikers) [inline recipes/bzip3/all/test_package/test_package.c:7]: `const`?

File: recipes/bzip3/all/CMakeLists.txt (hunk 2 above)
Comment (toge) [reply recipes/bzip3/all/CMakeLists.txt]: Thank you for review.
I missed this feature. I fixed it.
Please review again.

**Review by jwillikers — APPROVED** [review]
Awesome! Thanks!

**Review by SSE4 — DISMISSED** [review]

File: recipes/bzip3/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+    def package(self):
+        self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
+        cmake = self._configure_cmake()
+        cmake.install()
+
+    def package_info(self):
+        self.cpp_info.libs = ["bzip3"]
+        if self.settings.os in ["Linux", "FreeBSD"] and self.options.get_safe("with_thread", False):
+            self.cpp_info.system_libs.append("pthread")
```
Comment (prince-chrismc) [inline recipes/bzip3/all/conanfile.py:75]:
```suggestion
        if self.settings.os in ["Linux", "FreeBSD"]:
            self.cpp_info.system_libs.append("m")
            if self.options.get_safe("with_thread", False):
                self.cpp_info.system_libs.append("pthread")
```

**Review by prince-chrismc — COMMENTED** [review]
Maybe we can sneak in the missing system lib?

> Hooks produced the following warnings for commit [ffd8b97](https://github.com/conan-io/conan-center-index/commit/ffd8b97d5fe413f43c8ada423f11ca9fd9646869)
> bzip3/1.1.4
> ```
> post_package(): WARN: [MISSING SYSTEM LIBS (KB-H043)] Library './lib/libbzip3.so' links to system library 'm' but it is not in cpp_info.system_libs.
> ```

**Review by SSE4 — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

File: recipes/bzip3/all/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    source_subfolder/src/rle.c
+)
+target_include_directories(bzip3 PUBLIC source_subfolder/include)
+set_target_properties(bzip3 PROPERTIES
+    PUBLIC_HEADER "${BZIP3_INC}"
+    WINDOWS_EXPORT_ALL_SYMBOLS ON
+    C_EXTENSIONS OFF
+)
+if (with_thread)
+    target_compile_definitions(bzip3 PRIVATE "PTHREAD")
```
Comment (SpaceIm) [inline recipes/bzip3/all/CMakeLists.txt:30]: you should add:

```cmake
find_package(Threads REQUIRED)
target_link_libraries(bzip3 PRIVATE Threads::Threads)
```

File: recipes/bzip3/all/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+)
+
+if (WITH_UTIL)
+    install(
+        TARGETS bzip3_cmd
+        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
```
Comment (SpaceIm) [inline recipes/bzip3/all/CMakeLists.txt:55]: breaks iOS

should be:

```cmake
DESTINATION ${CMAKE_INSTALL_BINDIR}
```

File: recipes/bzip3/all/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.cppstd
+        del self.settings.compiler.libcxx
+        if self.settings.os not in ["Linux", "FreeBSD"]:
+            del self.options.with_thread
```
Comment (SpaceIm) [inline recipes/bzip3/all/conanfile.py:47]: - should be in `config_options()`
- why only Linux & FreeBSD? Should be deleted for Windows only (and actually it should work for MinGW with pthread)

File: recipes/bzip3/all/conanfile.py [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+            tools.patch(**patch)
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
+        cmake = self._configure_cmake()
+        cmake.install()
+
+    def package_info(self):
```
Comment (SpaceIm) [inline recipes/bzip3/all/conanfile.py:72]: you should add:

```python
        self.cpp_info.set_property("pkg_config_name", "bzip3")
```

see https://github.com/kspalaiologos/bzip3/blob/1.1.4/bzip3.pc.in

**Pull request merged**